src/Compilers/CSharp/Test/Semantic/Semantics/AnonymousFunctionTests.cs

```diff
@@ -0,0 +1,802 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
```

Comment:
>UnitTests [](start = 40, length = 9)

consider adding some tests that execute the code (CompileAndVerify with expectedOutput). Also, please include a test with `static async` or `async static`.